src/hobbits-plugins/operators/Edit/edit.cpp

```diff
@@ -0,0 +1,236 @@
+        qint64 outputSize = inputContainerSize - (length * 8) + (newBitsInRangeLength * 8);
+
+        QSharedPointer<const BitArray> bits = inputContainers.at(0)->bits();
+
+        if (newBitsInRangeLength == length) {     
+            m_outBits->setBytes(start, newAsciiInRange, 0, newBitsInRangeLength);
+        } else {
+            QSharedPointer<BitArray> postBits = QSharedPointer<BitArray>(new BitArray(outputSize - start + length ));
+            QByteArray byteArray = bits->readBytes(start + length, outputSize - 1);
+            int endLength = byteArray.length();
+            const char* data = byteArray.data();
+            m_outBits->setBytes(start+newBitsInRangeLength, data, 0, endLength);
+            m_outBits->setBytes(start, newAsciiInRange, 0, newBitsInRangeLength);
+            
+        }
+
+        */
+
+}
+
+QSharedPointer<const OperatorResult> Edit::operateOnBits(
+    QList<QSharedPointer<const BitContainer> > inputContainers,
+    const Parameters &parameters,
+    QSharedPointer<PluginActionProgress> progress)
+{    
+    if (inputContainers.length() == 0) {
+        QList<QSharedPointer<BitContainer>> outputContainers;
+        qint64 outputSize = 10;
+        QSharedPointer<BitArray> outBits;
+        outBits = QSharedPointer<BitArray>(new BitArray(outputSize));
+
+        auto bitContainer = BitContainer::create(outBits);
+
+        outputContainers.append(bitContainer);
+        return OperatorResult::result(outputContainers, parameters);
+    }
+
+    if (inputContainers.length() > 1) {
+        return OperatorResult::error("Requires a single input bit container");
+        
+    }
+
+    QStringList invalidations = m_delegate->validate(parameters);
+    if (!invalidations.isEmpty()) {
+        return OperatorResult::error(QString("Invalid parameters passed to %1:\n%2").arg(name()).arg(invalidations.join("\n")));
+    }
+
+    int progressInt = 15;
+
+    progress->setProgressPercent(progressInt);
+
+    //get values from spin boxes
+    qint64 start = parameters.value("start").toInt();
+    qint64 length = parameters.value("length").toInt();
+
+
+    //get bits from text box
+    QString newBitsInRange = parameters.value("new_bits_in_range").toString();
+
+    //pte_bits is already filled with data from start to length
+    //
+    //auto bitContainer = inputContainers.first();
+    QList<QSharedPointer<BitContainer>> outputContainers;
+
+    
+
+    
+
+    //do something with bits here
+    //ex 0000 1111 -> 1111 0000
+
+    int end = start + length;
+    qint64 inputContainerSize = inputContainers.at(0)->size();
+    QSharedPointer<const BitArray> bits = inputContainers.at(0)->bits();
+
+    progressInt+=15; progress->setProgressPercent(progressInt);
+
+    int newBitsInRangeLength = newBitsInRange.length();
+
+    int unitSize = getUnitSize(parameters);
+
+    qint64 outputSize = inputContainerSize - (length * unitSize) + (newBitsInRangeLength * unitSize);
```

Comment:
once you have unit size, it is probably best to just create pre-multiplied variables like `destBitLength`, `destBitOffset`, etc. You can also create the parsed `BitArray` and use that length explicitly rather than using `newBitsInRangeLength`